src/number/NumberTypes.jl

```diff
@@ -317,6 +322,71 @@ n_transExt(c::N_FField, n::Nemo.fmpz) = n_transExt(c, libSingular.number_ptr(n,
 n_transExt(c::N_FField, n::n_Z) = n_transExt(c, BigInt(n))
 
 
+###############################################################################
+#
+#   SingularAlgebraicExtensionField/n_algExt
+#
+###############################################################################
+
+const N_AlgExtFieldID = Dict{Tuple{N_FField, n_transExt}, Field}()
+const N_AlgExtFieldIDptr = Dict{libSingular.coeffs_ptr, Field}()
```

Comment:
@thofma could you take a look a my caching here?

Reply:
Do we need to always cache it?

Reply:
I have no idea. I think they should be cached be default, and the ones that come from singular should be cached. But, none of the other constructors have a cached argument. I should add the cached = keyword argument?

Reply:
Usually the user facing ones have the option to disable the caching. I guess some constructors here must have the caching to make it work with Singular? I don't know enough about the details.

Reply:
I added cached for the ones that face the user